src/main/java/com/project/financetracker/controller/FinancetrackerController.java

```diff
@@ -23,63 +25,76 @@ public class FinancetrackerController {
     private final TransactionService transactionService;
 
     @GetMapping("/")
-    public String helloWorld() {
-        return "Hello World!";
+    public ResponseEntity<String> helloWorld() {
+        return ResponseEntity
+                .status(HttpStatus.OK)
+                .body("Hello, world!");
     }
 
     @GetMapping("/test")
-    public String testEndpoint() {
-        return "Test is working!";
+    public ResponseEntity<String> testEndpoint() {
+        return ResponseEntity
+                .status(HttpStatus.OK)
+                .body("Test is working!");
     }
 
     @GetMapping("/register_user")
-    public String registerUser(@RequestParam(name = "un") String un,
-                               @RequestParam(name = "email") String mail,
-                               @RequestParam(name = "fn") String fn,
-                               @RequestParam(name = "ln") String ln) {
+    public ResponseEntity<String> registerUser(@RequestParam(name = "un") String un,
+                                               @RequestParam(name = "email") String mail,
+                                               @RequestParam(name = "fn") String fn,
+                                               @RequestParam(name = "ln") String ln) {
         UserInfo parsed = UserInfo.builder().username(un).email(mail).firstName(fn).lastName(ln).build(); // build the object
         UserInfo resp = registerService.registerUser(parsed); // pass in the object
 
         if (resp == null) {
-            return "Invalid input.";
```

Comment:
should we make custom exception for the different reasons, like user id is in use, invalid email ...,?